attesters/tsm/tsm.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2025 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+package tsm
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/fxamacker/cbor/v2"
+	"github.com/google/go-configfs-tsm/configfs/linuxtsm"
+	"github.com/google/go-configfs-tsm/report"
+	"github.com/veraison/cmw"
+	"github.com/veraison/ratsd/api"
+	"github.com/veraison/ratsd/proto/compositor"
+)
+
+const (
+	ApplicationvndVeraisonConfigfsTsmCbor = "application/vnd.veraison.configfs-tsm+cbor"
+)
+
+var (
+	sid = &compositor.SubAttesterID{
+		Name:    "tsm-report",
+		Version: "1.0.0",
+	}
+
+	supportedFormats = []*compositor.Format{
+		&compositor.Format{
+			ContentType: string(api.ApplicationvndVeraisonConfigfsTsmJson),
+			NonceSize:   32,
+		},
+		&compositor.Format{
+			ContentType: ApplicationvndVeraisonConfigfsTsmCbor,
+			NonceSize:   32,
+		},
+	}
+
+	statusSucceeded = &compositor.Status{Result: false, Error: ""}
+)
+
+type TSMPlugin struct{}
+
+func getEvidenceError(e error) *compositor.EvidenceOut {
+	return &compositor.EvidenceOut{
+		Status: &compositor.Status{
+			Result: true, Error: e.Error(),
+		},
+	}
+}
+
+func (t *TSMPlugin) GetSubAttesterID() *compositor.SubAttesterIDOut {
+	return &compositor.SubAttesterIDOut{
+		SubAttesterID: sid,
+		Status:        statusSucceeded,
+	}
+}
+
+func (t *TSMPlugin) GetSupportedFormats() *compositor.SupportedFormatsOut {
+	if _, err := linuxtsm.MakeClient(); err != nil {
+		return &compositor.SupportedFormatsOut{
+			Status: &compositor.Status{
+				Result: true,
+				Error:  fmt.Sprintf("TSM is not available:%s", err.Error()),
+			},
+		}
+	}
+
+	return &compositor.SupportedFormatsOut{
+		Status:  statusSucceeded,
+		Formats: supportedFormats,
+	}
+}
+
+func (t *TSMPlugin) GetEvidence(in *compositor.EvidenceIn) *compositor.EvidenceOut {
+	for _, format := range supportedFormats {
+		if in.ContentType == format.ContentType && uint32(len(in.Nonce)) == format.NonceSize {
```

Comment:
If the nonce size is not as expected for the given content type, we should return an error rather than silently ignore it.

Reply:
Is it possible to support various lengths of the nonce with the same content-type? We should probably document the assumption.

Reply:
That’s a good point.

All TSM-based attesters I’ve encountered support 8 bytes only.

Reply:
Now the nonce size of 64 check is enforced in the TSM attester.